src/Raven.Server/Utils/TypeConverter.cs

```diff
@@ -292,6 +319,9 @@ private static void NestingLevelTooDeep(object value)
 
         private static IEnumerable<object> Flatten(IEnumerable items)
         {
+            if (items == null)
+                EnumerableNullException();
```

Comment:
I think that passing `DynamicNullObject` would make it will throw but it shouldn't. It's a bit dangerous, I would drop that check to avoid possible regression bug in indexing.

Reply:
Please check tests: http://jenkins.hibernatingrhinos.com/job/PR_Tests_v4.1/2195/testReport/

I'm pretty sure they failed due to this check

Reply:
done